src/framework/ui/internal/platform/linux/linuxplatformtheme.cpp

```diff
@@ -28,26 +28,53 @@
 using namespace mu::ui;
 using namespace mu::async;
 
+static const QLatin1String service("org.freedesktop.portal.Desktop");
+static const QLatin1String path("/org/freedesktop/portal/desktop");
+static const QLatin1String interface("org.freedesktop.portal.Settings");
+
 LinuxPlatformTheme::LinuxPlatformTheme()
 {
 }
 
 void LinuxPlatformTheme::startListening()
 {
+    QDBusConnection::sessionBus().connect(service, path, interface, QStringLiteral("SettingChanged"), this,
+                                          SLOT(processSettingChange(QString,QString,QDBusVariant)));
+    m_isSystemThemeDark = isSystemThemeCurrentlyDark();
+    m_isListening = true;
 }
 
 void LinuxPlatformTheme::stopListening()
 {
+    QDBusConnection::sessionBus().disconnect(service, path, interface, QStringLiteral("SettingChanged"), this,
+                                             SLOT(processSettingChange(QString,QString,QDBusVariant)));
+    m_isListening = false;
 }
 
 bool LinuxPlatformTheme::isFollowSystemThemeAvailable() const
 {
-    return false;
+    const QDBusConnection connection = QDBusConnection::sessionBus();
+
+    static const QString registrarService = QStringLiteral("org.freedesktop.portal.Desktop");
```

Comment:
Can't this be `service`? Or is it really a different thing with a different role, that only coincidentally has the same value as `service`?

Reply:
> Can't this be `service`? Or is it really a different thing with a different role, that only coincidentally has the same value as `service`?

I didn't notice that, I'll fix it...
Both strings are used for the same D-Bus service.